### DISCUSSION TITLE
fail to search on all shard leaders

### DISCUSSION THREAD

**Discussion by RongHouchao** [milvus-io/milvus#20395] [Q&A and General discussion]
| 2022-11-08 12:59:50.707  INFO 13210 --- [query_record_callable_4] p.r.i.VectorTbIndexFeatureRepositoryImpl : queryByVector.response: R{exception=Exception: fail to search on all shard leaders, err=All attempts results:
attempt #1:fail to get shard leaders from QueryCoord: no replica available
attempt #2:fail to get shard leaders from QueryCoord: no replica available
attempt #3:fail to get shard leaders from QueryCoord: no replica available
attempt #4:fail to get shard leaders from QueryCoord: no replica available
attempt #5:fail to get shard leaders from QueryCoord: no replica available
attempt #6:fail to get shard leaders from QueryCoord: no replica available
attempt #7:fail to get shard leaders from QueryCoord: no replica available
attempt #8:context deadline exceeded
, status=1, data=null}

向量召回的时候报了这个错误，这是什么错误

**Comment by RongHouchao** [comment]
![image](https://user-images.githubusercontent.com/117245915/200482429-84f3c2e4-679b-42b1-b289-f8ea84d2d3e5.png)

**Comment by yhmo** [comment]
Similar to this issue: https://github.com/milvus-io/milvus/issues/20157

What is the workflow before you got this error?

If you deployed milvus by standalone, just use **`docker logs [container id]`** to export the log. If you deployed milvus by k8s, follow this [script](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export the log.

**Reply by RongHouchao** [reply]
What is the workflow before you got this error?
---- 之前使用一直是好的，昨天系统盘满了，然后把/var/lib/docker/containers目录下milvus-standalone的json.log 删除了。重启了服务器，重启了milvus服务，然后再使用就这样了。我使用的单机版的milvus v2.1.0版本。

**Reply by yhmo** [reply]
> What is the workflow before you got this error? ---- 之前使用一直是好的，昨天系统盘满了，然后把/var/lib/docker/containers目录下milvus-standalone的json.log 删除了。重启了服务器，重启了milvus服务，然后再使用就这样了。我使用的单机版的milvus v2.1.0版本。

2.1.0有不少易现的稳定性问题，2.1.4想对来说好一些。马上我们要发布2.2，可以考虑换新版本。

**Reply by RongHouchao** [reply]
> > What is the workflow before you got this error? ---- 之前使用一直是好的，昨天系统盘满了，然后把/var/lib/docker/containers目录下milvus-standalone的json.log 删除了。重启了服务器，重启了milvus服务，然后再使用就这样了。我使用的单机版的milvus v2.1.0版本。
> 
> 2.1.0有不少易现的稳定性问题，2.1.4想对来说好一些。马上我们要发布2.2，可以考虑换新版本。


基本都是发生在数据量比较大的情况下，都不敢重启。
现在4核32G机器，存入200万数据，大概率会起不来。
2.1.0可以直接升级到2.1.4吗？有操作手册的链接么。

**Reply by yhmo** [reply]
> > > What is the workflow before you got this error? ---- 之前使用一直是好的，昨天系统盘满了，然后把/var/lib/docker/containers目录下milvus-standalone的json.log 删除了。重启了服务器，重启了milvus服务，然后再使用就这样了。我使用的单机版的milvus v2.1.0版本。
> > 
> > 
> > 2.1.0有不少易现的稳定性问题，2.1.4想对来说好一些。马上我们要发布2.2，可以考虑换新版本。
> 
> 基本都是发生在数据量比较大的情况下，都不敢重启。 现在4核32G机器，存入200万数据，大概率会起不来。 2.1.0可以直接升级到2.1.4吗？有操作手册的链接么。

2.1.0主要是有超大内容写入etcd导致启动时读取元数据失败的问题，后来在2.1.2/2.1.4上基本修复。
2.1.4主要是querycoord启动时恢复流程调度上有问题，有可能造成重启时卡死。这个问题在2.2上通过重构了querycoord的内部逻辑来修复。一旦出现就不太好办，有个dev工具有可能可以帮助恢复，但不能百分百保证：https://github.com/milvus-io/birdwatcher。一般如果用户的数据有备份的话，不如直接重新导入。

2.1.0升到2.1.4可以直接升：停止旧的服务，下载2.1.4的docker image，用相同的volume来启动2.1.4服务。
